### PR TITLE
Added an autocomplete input for grouping links

### PR DESCRIPTION
**Pull request by Elijah-trillionz**
## Fixes Issue

Closes: #8343

## Changes proposed

- [X] Fetched all existing links of user to get existing groups of user (without any additional delay in loading the UI)
- [x] Used [headless Combobox](https://headlessui.com/react/combobox) to render the existing groups in an autocomplete input

## Check List (Check all the applicable boxes) 

- [X] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] All new and existing tests passed.
- [X] This PR does not contain plagiarized content.
- [X] The title of my pull request is a short description of the requested changes.

## Screenshots
linkfree-solution1-pr

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/8428"></a>

### CODE REVIEW

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for the contribution 👍 

- the changes to the package lock file are not required 
- we already have a dropdown component, can that be reused?

**Comment by Elijah-trillionz** [comment]
> * the changes to the package lock file are not required

I didn't make any changes to the package lock file.
It was during pulling to update with the current linkfree main branch (as at the time of creating this pr) that it the version of linkfree got updated.

**Comment by Elijah-trillionz** [comment]
> * we already have a dropdown component, can that be reused?

The Select component (which recently replaced the DropDown component) because it is a select input. and I can't reuse it for an autocomplete input.

**Comment by eddiejaoude** [comment]
- Did you mean to close this PR?
- Please explain why you can reuse it?

**Comment by Elijah-trillionz** [comment]
I didn't mean to close it, I had to discard my remote changes so as to sync to the current version of the linkfree on my local devices. I still have the changes it on my PC.

Secondly, the Select component is a select input like I said before. With the newest updates to LinkFree, there isn't anymore Dropdown component as that has been replaced with the Select component.

The Combobox component I'm using is from headless/ui, and it is an autocomplete component with an input and a dropdown (customizable).

The select component only supports selecting. Using this means, the user must have existing groups (first) can't type in a new group. But what I need is an autocomplete with the ability to create a non-existing value while being able to select from a set of values.

File: components/form/InputDropDown.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+              key={index}
+              value={_group}
+              className={`cursor-pointer`}
+            >
+              {({ active, selected }) => (
+                <span
+                  className={`${
+                    active ? "bg-secondary-low dark:bg-secondary-high" : ""
+                  } ${
+                    selected
```
Comment (azhechkova) [inline components/form/InputDropDown.js:67]: You can use `classNames` util for more readable classes, you can find it at `services/utils/classNames.js`

File: package-lock.json [hunk 2]
```diff
@@ -6,7 +6,11 @@
   "packages": {
     "": {
       "name": "linkfree",
```
Comment (azhechkova) [inline package-lock.json:8]: Fix conflict in package-lock or remove the changes, please

File: components/form/InputDropDown.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+              className="h-5 w-5 text-gray-400"
+              aria-hidden="true"
+            />
+          </Combobox.Button>
+        </div>
+        <Combobox.Options
+          className="bg-white absolute border-2 transition-all duration-250 ease-linear rounded mb-2 block w-full 
+        dark:bg-primary-high focus:ring-0 focus:border-tertiary-medium focus:outline-0 border-tertiary-medium"
+        >
+          {filteredLinkGroups.map((_group, index) => (
```
Comment (azhechkova) [inline components/form/InputDropDown.js:56]: Not sure why we need an underscore for `group` var

**Comment by Elijah-trillionz** [comment]
Thought this community would be a good place to start open-source contributing proper. But I was wrong.
It's the worst place

**Comment by eddiejaoude** [comment]
Sorry to hear that @Elijah-trillionz , please can you explain why? I know I am behind on notifications but I am trying my best 😢

**Pull request closed without merging**